Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair_degeneracies.h

```diff
@@ -1696,7 +1696,9 @@ bool remove_degenerate_faces(const FaceRange& face_range,
   {
     std::map<edge_descriptor, bool> are_degenerate_edges;
 
-    for(face_descriptor fd : degenerate_face_set)
+    // use a copy of the set to avoid iterator invalidation
+    std::set<face_descriptor> degenerate_face_set_copy(degenerate_face_set);
+    for(face_descriptor fd : degenerate_face_set_copy)
```

Comment:
Even better, copy in a vector, to avoid the multiple memory allocations:
```suggestion
    std::vector<face_descriptor> degenerate_face_set_copy(degenerate_face_set.size());
    std::copy(degenerate_face_set.begin(), degenerate_face_set.end(),
              degenerate_face_set_copy.begin());
    for(face_descriptor fd : degenerate_face_set_copy)
```